tests/baselines/reference/functionExpressionNames.types

```diff
@@ -31,7 +31,7 @@ var o = {
         this.c = 'nested object'
 >this.c = 'nested object' : "nested object"
 >this.c : any
->this : this
+>this : { C: typeof C; }

```

Comment:
This test is explicitly the case that is changing:

```js
var o = {
    C: function () {
        this.c = 'nested object'
    }
}
var og = new o.C();
```

So, I'm not sure if that's good or bad.